great_expectations/checkpoint/actions.py

```diff
@@ -85,23 +86,42 @@ def _build_renderer(config: dict) -> Renderer:
     return renderer
 
 
+class ValidationActionRunStatus(enum.Enum):
+    SUCCESS = "success"
+    FAILURE = "failure"
+    NOT_RUN = "not_run"
+
+
+class ValidationActionResult(BaseModel):
+    """
+    The result of running a ValidationAction.
+
+    Args:
+        status: The status of the action run. If the action was not run, the status is "not_run".
+        run_info: A dictionary containing information about the run.
+    """
+
+    status: ValidationActionRunStatus
```

Comment:
Should we include the action name (eg the class name) in this result object or will it be obvious since this is embedded inside something? Eg if I see `not_run` will I know that is associated with a pager duty action?

Reply:
It seems like the run_info dict has a key which is a string like `<action_type>_result`. If we put the action type in it's own field, it would make that info easier to parse out.

Reply:
It'll be obvious since this is only ever used in `Checkpoint.run` - my thinking is we can take the aggregated results (of these actions) and either log or append to the checkpoint result object.

Don't want to make too many changes here until we can plan better.